@xen-orchestra/rest-api/src/rest-api/rest-api.mts

```diff
@@ -0,0 +1,11 @@
+import type { XoApp } from './rest-api.type.mjs'
+
+export class RestApi {
+  authenticateUser: XoApp['authenticateUser']
+  runWithApiContext: XoApp['runWithApiContext']
+
+  constructor(xoApp: XoApp) {
+    this.authenticateUser = (credentials, userData, opts) => xoApp.authenticateUser(credentials, userData, opts)
+    this.runWithApiContext = (user, cb) => xoApp.runWithApiContext(user, cb)
```

Comment:
would it be possible to define this as real method of the RestApi class ?

Reply:
I am not sure to understand  what you're waiting for

Reply:
You mean to declare them outside of the constructor?
```
class RestApi {
  constructor(){}
  
  authenticateUser(){...}
}
```

If so, since all these methods depend of `xoApp`, that means we'll need to keep an instance of `xoApp` in the `RestApi` class. Since `xoApp` is VERY big, I'd like to avoid keeping it in memory here for "nothing"

Reply:
a reference to an object don't cost anything, only one reference. Also having a real method means a type, and code completion